### PR TITLE
Sscs 7994 joint party added

### PR DESCRIPTION
**Pull request by chrisdavecm**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/SSCS-7994

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/e2e/java/uk/gov/hmcts/reform/sscs/functional/DwpUploadFunctionalTest.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        json = json.replace("DWP_AT38_DOCUMENT_URL", documentUrl);
+        json = json.replace("DWP_AT38_BINARY_DOCUMENT_URL", binaryUrl);
+        json = json.replace("DWP_RESPONSE_DOCUMENT_URL", documentUrl);
+        json = json.replace("DWP_RESPONSE_BINARY_DOCUMENT_URL", binaryUrl);
+        json = json.replace("DWP_EVIDENCE_DOCUMENT_URL", documentUrl);
+        json = json.replace("DWP_EVIDENCE_BINARY_DOCUMENT_URL", binaryUrl);
+
+        simulateCcdCallback(json);
+
+        System.out.println("ccdCaseId" + ccdCaseId);
```
Comment (ryandac) [inline src/e2e/java/uk/gov/hmcts/reform/sscs/functional/DwpUploadFunctionalTest.java:54]: remove/ replace println

**Review by ryandac — COMMENTED** [review]
looks okay apart from the println

File: src/main/java/uk/gov/hmcts/reform/sscs/callback/handlers/DwpUploadResponseHandler.java [hunk 2]
```diff
@@ -115,14 +117,19 @@ private void handleUc(Callback<SscsCaseData> callback) {
...
-
             caseData.setDwpState(DwpState.RESPONSE_SUBMITTED_DWP.getId());
 
             ccdService.updateCase(caseData, callback.getCaseDetails().getId(),
                 EventType.DWP_RESPOND.getCcdType(), "Response received",
                 description, idamService.getIdamTokens());
         }
+
+        if (caseData.getJointParty() != null && StringUtils.equalsIgnoreCase("Yes", caseData.getJointParty())) {
+            ccdService.updateCase(caseData, callback.getCaseDetails().getId(),
```
Comment (jackmaloney) [inline src/main/java/uk/gov/hmcts/reform/sscs/callback/handlers/DwpUploadResponseHandler.java:128]: Why does this need to be a CCD event?

File: src/main/java/uk/gov/hmcts/reform/sscs/callback/handlers/DwpUploadResponseHandler.java [hunk 3]
```diff
@@ -115,14 +117,19 @@ private void handleUc(Callback<SscsCaseData> callback) {
...
+                EventType.JOINT_PARTY_ADDED.getCcdType(), "Joint party added",
+                "description", idamService.getIdamTokens());
```
Comment (jackmaloney) [inline src/main/java/uk/gov/hmcts/reform/sscs/callback/handlers/DwpUploadResponseHandler.java:130]: Can you give the description a better sentence than just the word "description"?

File: src/main/java/uk/gov/hmcts/reform/sscs/callback/handlers/CaseUpdatedHandler.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        SscsCaseData caseData = callback.getCaseDetails().getCaseData();
+
+        BenefitType benefitType = caseData.getAppeal().getBenefitType();
+        log.info("BenefitType" + benefitType);
+
+        if (StringUtils.equalsIgnoreCase(benefitType.getCode(), "uc")) {
+            if (caseData.getJointParty() != null && StringUtils.equalsIgnoreCase("Yes", caseData.getJointParty())) {
+                ccdService.updateCase(caseData, callback.getCaseDetails().getId(),
+                    EventType.JOINT_PARTY_ADDED.getCcdType(), "Joint party added",
+                    "description", idamService.getIdamTokens());
```
Comment (ryandac) [inline src/main/java/uk/gov/hmcts/reform/sscs/callback/handlers/CaseUpdatedHandler.java:63]: empty string is better than the word description

**Review by ryandac — APPROVED** [review]

**Pull request merged**